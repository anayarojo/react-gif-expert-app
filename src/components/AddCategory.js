import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (event) => {
        setInputValue(event.target.value);
    }

    const handleSubmit = (event) => {
        event.preventDefault();

        if (inputValue.trim() !== '') {
            setCategories((categories) => [inputValue, ...categories]);
            setInputValue('');   
        }
    }

    return (
        <form className="mb-5" onSubmit={ handleSubmit }>
            <input 
                type="text" 
                value={ inputValue }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
};

export default AddCategory;
