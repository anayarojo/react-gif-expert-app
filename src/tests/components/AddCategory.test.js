import React from 'react';
import { shallow } from 'enzyme'
import '@testing-library/jest-dom';

import AddCategory from '../../components/AddCategory';

describe('Test for <AddCategory />', () => {

    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={ setCategories } />);

    beforeEach(() => {

        jest.clearAllMocks();
        wrapper = shallow(<AddCategory setCategories={ setCategories } />);
    });

    test('Should show <AddCategory /> successfully', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('Should to change text input', () => {

        const value = 'Goku';
        let input = wrapper.find('input');
        
        input.simulate('change', { target: { value } });
        input = wrapper.find('input');

        expect(input.prop('value')).toBe(value);
    });

    test('Should not form submit', () => {

        const form = wrapper.find('form');
        form.simulate('submit', { preventDefault(){} });

        expect( setCategories ).not.toHaveBeenCalled();
    });

    test('Should call setCategories() and clean text input', () => {

        const value = 'Naruto';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate('submit', { preventDefault(){} });

        expect(setCategories).toHaveBeenCalled();
        expect(setCategories).toHaveBeenCalledTimes(1);
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
        
        expect(wrapper.find('input').prop('value')).toBe('');
    });

});
