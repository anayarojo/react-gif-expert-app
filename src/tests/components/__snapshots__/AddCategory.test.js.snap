// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for <AddCategory /> Should show <AddCategory /> successfully 1`] = `
<form
  className="mb-5"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
